src/ledger/LedgerTxnAccountSQL.cpp

```diff
@@ -330,4 +330,70 @@ LedgerTxnRoot::Impl::dropAccounts()
         << "CREATE INDEX accountbalances ON accounts (balance) WHERE "
            "balance >= 1000000000";
 }
+
+static std::unordered_map<std::string, std::string>
+loadHomeDomainsToEncode(Database& db, size_t limit, size_t offset)
+{
+    std::string accountID, homeDomain;
+
+    std::string sql = "SELECT accountid, homedomain FROM accounts "
+                      "ORDER BY accountid LIMIT :v1 OFFSET :v2";
+
+    auto prep = db.getPreparedStatement(sql);
+    auto& st = prep.statement();
+    st.exchange(soci::into(accountID));
+    st.exchange(soci::into(homeDomain));
+    st.exchange(soci::use(limit));
+    st.exchange(soci::use(offset));
+    st.define_and_bind();
+    st.execute(true);
+
+    std::unordered_map<std::string, std::string> res;
```

Comment:
nit: you don't really need a k:v type of structure for this (and return type of the function), a `std::vector<std::pair<>>` is enough